client/app/bundles/course/assessment/question/programming/components/ProgrammingQuestionForm.jsx

```diff
@@ -0,0 +1,339 @@
+import React, { PropTypes } from 'react';
+import ReactSummernote from 'react-summernote';
+import Select from 'react-select';
+import { injectIntl, defineMessages } from 'react-intl';
+import styles from './ProgrammingQuestionForm.scss'
+
+import 'react-select/dist/react-select.css';
+
+const translations = defineMessages({
+  attemptLimitPlaceholderMessage: {
+    id: 'course.assessment.question.programming.programmingQuestionForm.attemptLimitPlaceholderMessage',
+    defaultMessage: 'The maximum times that the students can test their answers (does not apply to staff)',
+    description: 'Placeholder message for attempt limit input field.',
+  },
+  downloadPackage: {
+    id: 'course.assessment.question.programming.programmingQuestionForm.downloadPackage',
+    defaultMessage: 'Download Package',
+    description: 'Title for downloading the zip package.',
+  },
+  editTestsOnlineButton: {
+    id: 'course.assessment.question.programming.programmingQuestionForm.editTestsOnlineButton',
+    defaultMessage: 'Edit Tests Online',
+    description: 'Button for editing tests online.',
+  },
+  uploadPackageButton: {
+    id: 'course.assessment.question.programming.programmingQuestionForm.uploadPackageButton',
+    defaultMessage: 'Upload Package',
+    description: 'Button for uploading package.',
+  },
+  submitButton: {
+    id: 'course.assessment.question.programming.programmingQuestionForm.submitButton',
+    defaultMessage: 'Submit',
+    description: 'Button for submitting the form.',
+  },
+  evaluatingMessage: {
+    id: 'course.assessment.question.programming.programmingQuestionForm.evaluatingMessage',
+    defaultMessage: 'Evaluating',
+    description: 'Text to be displayed when evaluating the programming question.',
+  },
+  loadingMessage: {
+    id: 'course.assessment.question.programming.programmingQuestionForm.loadingMessage',
+    defaultMessage: 'Loading',
+    description: 'Text to be displayed when waiting for server response after form submission.',
+  },
+});
+
+const propTypes = {
+  data: React.PropTypes.shape({
+    question: PropTypes.object.isRequired,
+    formData: PropTypes.object.isRequired,
+    isLoading: PropTypes.bool.isRequired,
+    isEvaluating: PropTypes.bool.isRequired
+  }),
+  actions: React.PropTypes.shape({
+    updateProgrammingQuestion: PropTypes.func.isRequired,
+    updateSkills: PropTypes.func.isRequired,
+    updateEditorMode: PropTypes.func.isRequired
+  }),
+  intl: PropTypes.shape({
+    formatMessage: PropTypes.func.isRequired,
+  }).isRequired,
+};
+
+class ProgrammingQuestionForm extends React.Component {
+
+  componentWillReceiveProps(nextProps) {
+    this.isLoading = nextProps.data.isLoading;
+    this.isEvaluating = nextProps.data.isEvaluating;
```

Comment:
Is this needed? Possible to do without the assignment and check `this.props.isLoading` and `this.props.isEvaluating` directly?